packages/components/src/components/dropdown-menu/DropdownMenu.tsx

```diff
@@ -92,14 +89,14 @@ export const DropdownMenu: React.FC<DropdownMenuProps> = ({children, className,
 
   const mouseEventHandler = (e: MouseEvent) => {
     e.stopPropagation();
-    if (onClose && menu && !(e.composedPath() as any).includes(menu.current)) {
+    if (onClose && ref && !(e.composedPath() as any).includes(ref.current)) {
       onClose();
     }
   }
 
   const removeListeners = useCallback(() => {
-    window.removeEventListener('keyup', keyboardEventHandler);
-    window.removeEventListener('click', mouseEventHandler);
+    ref.current?.ownerDocument.removeEventListener('keyup', keyboardEventHandler);
```

Comment:
The `ref.current?` is added because current can be undefined but the React 16 types don't reflect that. Another good reason to upgrade to React 17.